AppDB/backup/backup_recovery_helper.py

```diff
@@ -13,6 +45,81 @@ def does_file_exist(path):
   """
   return os.path.isfile(path)
 
+def enough_disk_space(service):
+  """ Checks if there's enough available disk space for a new backup.
+
+  Returns:
+    True on success, False otherwise.
+  """
+  available_space = get_available_disk_space()
+  logging.debug("Available space: {0}".format(available_space))
+
+  backup_size = get_backup_size(service)
+  logging.debug("Backup size: {0}".format(backup_size))
+
+  if backup_size > available_space * \
+    backup_recovery_constants.PADDING_PERCENTAGE:
+    logging.warning("Not enough space for a backup.")
+    return False
+  return True
+
+def get_available_disk_space():
+  """ Returns the amount of available disk space under /opt/appscale.
+
+  Returns:
+    An int, the available disk space in bytes.
+  """
+  stat_struct = os.statvfs(os.path.dirname(BACKUP_DIR_LOCATION))
+  return stat_struct[statvfs.F_BAVAIL] * stat_struct[statvfs.F_BSIZE]
+
+def get_backup_size(service):
+  """ Sums up the size of the snapshot files that consist the backup for the
+  given service.
+
+  Args:
+    service: A str, the service for which we'll calculate the backup size.
+  Returns:
+    An int, the total size of the files consisting the backup in bytes.
+  """
+  backup_files = get_snapshot_paths(service)
+  total_size = sum(getsize(file) for file in backup_files)
+  return total_size
+
+def get_snapshot_paths(service):
+  """ Returns a list of file names holding critical data for the given service.
+
+  Args:
+    service: A str, the service for which we're getting the data files.
+    Currently there is support for Cassandra and Zookeeper.
+  Returns:
+    A list of full paths.
+  """
+  if service == 'cassandra':
```

Comment:
This probably doesn't need to be an if/else. Just return if !=. "look_for" doesn't need to be defined conditionally.